span-normalizer/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/jaeger/tenant/JaegerKeyBasedTenantIdProvider.java

```diff
@@ -24,14 +26,25 @@ public JaegerKeyBasedTenantIdProvider(String tenantIdKey) {
     return Optional.of(tenantIdKey);
   }
 
+  /**
+   * gets the tenant id from process tags, with a fallback to getting it from the span tags.
+   *
+   * @param span
+   * @param tags
+   * @return the tenant id if it is present in the process tags or span tags; otherwise, returns an
+   *     empty optional.
+   */
   @Override
-  public Optional<String> getTenantId(Map<String, JaegerSpanInternalModel.KeyValue> tags) {
-    JaegerSpanInternalModel.KeyValue value = tags.get(tenantIdKey);
-    return Optional.ofNullable(value != null ? value.getVStr() : null);
+  public Optional<String> getTenantId(Span span, Map<String, KeyValue> tags) {
+    return span.getProcess().getTagsList().stream()
```

Comment:
There's logic to drop this key from the tags list - I'm not exactly sure why, but is that something we'd want on the resource too? It nags at me that it's inconsistent, but it also seems like we shouldn't have been dropping the attributes in the first place.

You can see where this is happening by following how `getTenantIdTagKey()` above gets used.

Reply:
> There's logic to drop this key from the tags list - I'm not exactly sure why, but is that something we'd want on the resource too?

I don't see any reason to drop this attribute from resource. That's why left it unchanged.

Reply:
consistency! If we're not dropping it from the resource, then we should remove the logic to drop it from the tags, too. If we think it's important to leave dropping it from the tags, then we should drop it from the resource, too. It's the same piece of information with two different transport mechanisms, so we want to preserve it (or not) consistently too.

Reply:
ok, since we're not sure if we should be dropping the tenant id from span tags or not, I'll keep it as it is (i.e. continue to drop it), and update to drop it from resources.

Reply:
Made this change.